src/zoo.ts

```diff
@@ -0,0 +1,69 @@
+// Інтерфейс Visitor, що описує основні властивості відвідувача
+interface Visitor {
+    name: string;
+    contactInfo: string;
+}
+
+// Інтерфейс Client, що наслідує властивості від Visitor
+interface Client extends Visitor {}
+
+// Клас Ticket для представлення квитка
+class Ticket {
+    constructor(public type: string, public price: number) {}
+}
+
+// Клас Visitor для створення об'єктів відвідувачів
+class Visitor {
+    constructor(public name: string, public contactInfo: string) {}
+}
+
+// Клас Cashier для управління продажем квитків і зберіганням інформації про відвідувачів
+class Cashier {
+    private currentVisitors: Visitor[] = []; // Список поточних відвідувачів
+    private clients: Client[] = []; // Список клієнтів
+    private revenue: number = 0; // Виручка
+
+    // Ціни на квитки залежно від їх типу
+    private ticketPrices: { [key: string]: number } = {
+        adult: 20,
+        child: 10,
+        family: 50
+    };
+
+    // Метод для продажу квитка
+    sellTicket(type: string, visitor: Visitor): void {
+        const price = this.ticketPrices[type];
+        if (price === undefined) {
+            throw new Error("Invalid ticket type.");
+        }
+
+        this.currentVisitors.push(visitor);
+        this.clients.push(visitor as Client);
+        this.revenue += price;
+    }
+
+    // Метод для сповіщення відвідувачів
+    notifyVisitors(message: string): void {
```

Comment:
Let's breathe life into our notification system by implementing the Observer pattern across all notification features! Let's make sure that everything that needs to be notified gets the message loud and clear.